src/server/game/BattlePets/BattlePetMgr.cpp

```diff
@@ -96,6 +98,15 @@ void BattlePetMgr::Initialize()
     if (QueryResult result = LoginDatabase.Query("SELECT MAX(guid) FROM battle_pets"))
         sObjectMgr->GetGenerator<HighGuid::BattlePet>().Set((*result)[0].GetUInt64() + 1);
 
+    for (BattlePetSpeciesEntry const* battlePetSpecies : sBattlePetSpeciesStore)
+    {
+        if (uint32 creatureId = battlePetSpecies->CreatureID)
+            _battlePetSpeciesByCreature[creatureId] = battlePetSpecies;
+
+        if (uint32 spellId = battlePetSpecies->SummonSpellID)
```

Comment:
You need to very carefully compare this code to the one using SummonProperties (client uses the old spellmgr one to determine learning battle pets)

Reply:
You're right: https://gist.github.com/meji46/f709eb5ecb4d0c19f89b27846ed14aaa I've restored the load from SpellMgr

Thanks 😃